src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/ItemManager.cs

```diff
@@ -759,5 +763,90 @@ private static bool IsPotentialFilterCharacter(char c)
+            int selectedIndex,
+            ImmutableArray<string> defaults,
+            bool aggressive)
+        {
+            // We only preselect when we are very confident with the selection, so don't override it.
+            if (defaults.IsDefaultOrEmpty || itemsWithMatch[selectedIndex].item.Rules.MatchPriority >= MatchPriority.Preselect)
+                return (selectedIndex, false);
+
+            if (aggressive)
+            {
+                foreach (var defaultText in defaults)
+                {
+                    for (var itemIndex = 0; (itemIndex < itemsWithMatch.Length); ++itemIndex)
+                    {
+                        var (item, patternMatch) = itemsWithMatch[itemIndex];
+                        if (item.FilterText == defaultText)
+                        {
+                            if (patternMatch == null || patternMatch.Value.Kind <= PatternMatchKind.Prefix)
+                                return (itemIndex, true);
+
+                            break;
+                        }
+                    }
+                }
+            }
+            else
+            {
+                int similarItemsStart;
+                int dissimilarItemIndex;
+
+                if (filterText.Length == 0)
+                {
+                    // If there is no applicableToSpan, then all items are equally similar.
+                    similarItemsStart = 0;
+                    dissimilarItemIndex = itemsWithMatch.Length;
+                }
+                else
+                {
+                    // Assume that the selectedIndex is in the middle of a range of -- as far as the pattern matcher is concerned --
+                    // equivalent items. Find the first & last items in the range and use that to limit the items searched for from
+                    // the defaults list.                    
+                    var (_, selectedItemMatch) = itemsWithMatch[selectedIndex];
+                    if (!selectedItemMatch.HasValue)
+                        return (selectedIndex, false);
+
+                    similarItemsStart = selectedIndex;
+                    while (--similarItemsStart >= 0)
+                    {
+                        var (_, itemMatch) = itemsWithMatch[similarItemsStart];
+                        if ((!itemMatch.HasValue) || itemMatch.Value.CompareTo(selectedItemMatch.Value) > 0)
+                            break;
+                    }
+
+                    similarItemsStart++;
+
+                    dissimilarItemIndex = selectedIndex;
+                    while (++dissimilarItemIndex < itemsWithMatch.Length)
+                    {
+                        var (_, itemMatch) = itemsWithMatch[dissimilarItemIndex];
+                        if ((!itemMatch.HasValue) || itemMatch.Value.CompareTo(selectedItemMatch.Value) > 0)
+                            break;
+                    }
+                }
+
+                if (dissimilarItemIndex > selectedIndex + 1)
```

Comment:
Should this be if (dissimilarItemIndex > similarItemStart + 1)
(this could have been a bug in the original code if you copied).

Reply:
Yes, this was copied from the original implementation, but I have since changed the code so this check is no longer there. Consider reviewing the diff between the first commit (code with original logic intact) and the last for additional changes I made.